src/main/java/com/spider/solomon/reporting/TrendAnalyzer.java

```diff
@@ -0,0 +1,51 @@
+package com.spider.solomon.reporting;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+import org.springframework.stereotype.Component;
+
+import com.spider.solomon.order.CustomerOrder;
+import com.spider.solomon.order.CustomerOrderDao;
+
+import jakarta.annotation.Resource;
+
+/**
+ * Analyzes trends in sales data
+ */
+@Component
+public class TrendAnalyzer {
+	@Resource
+	CustomerOrderDao customerOrderDao;
+
+	/**
+	 * Returns whether the total sales data is trending up or down
+	 */
+	public String saleAmountTrend () {
+		SimpleRegression r = new SimpleRegression();
+
+		LocalDateTime tmp = customerOrderDao.getAllOrders().stream()
+			.map(CustomerOrder::getOrderDate)
+			.min(LocalDateTime::compareTo)
+			.orElse(LocalDateTime.MIN);
+
+		// Add in data for all orders to the SimpleRegression
+		customerOrderDao.getAllOrders().forEach(order -> {
+			Long day = ChronoUnit.DAYS.between(tmp, order.getOrderDate());
+			double totalAmount = order.getTotalAmount().doubleValue();
+			r.addData(day, totalAmount);
```

Comment:
I'm wondering about the granularity of "day" for this -- it may make sense, but I'm curious.. If there is only one day's worth of orders, would the trend be irrelevant?